03_launch_mgmt_cluster.sh

```diff
@@ -66,55 +66,17 @@ function clone_repos() {
 # ------------------------------------
 
 #
-# Modifies the images to use the ones built locally in the kustomization
-# This is v1a3 specific for BMO, all versions for Ironic
-#
-function update_kustomization_images(){
-  FILE_PATH=$1
-  for IMAGE_VAR in $(env | grep "_LOCAL_IMAGE=" | grep -o "^[^=]*") ; do
-    IMAGE=${!IMAGE_VAR}
-    #shellcheck disable=SC2086
-    IMAGE_NAME="${IMAGE##*/}"
-    LOCAL_IMAGE="${REGISTRY}/localimages/$IMAGE_NAME"
-    OLD_IMAGE_VAR="${IMAGE_VAR%_LOCAL_IMAGE}_IMAGE"
-    # Strip the tag for image replacement
-    OLD_IMAGE="${!OLD_IMAGE_VAR%:*}"
-    if [[ $OLD_IMAGE == *"$CONTAINER_REGISTRY"* ]]; then
-      #shellcheck disable=SC2116
-      OLD_IMAGE="$(echo "${OLD_IMAGE/$CONTAINER_REGISTRY/$REPO_IMAGE_PREFIX}")"
-      echo "Image $OLD_IMAGE will be replaced with local image $LOCAL_IMAGE in image kustomization"
-    fi
-    sed -i -E "s $OLD_IMAGE$ $LOCAL_IMAGE g" "$FILE_PATH"
-  done
-  # Assign images from local image registry for kustomization
-  for IMAGE_VAR in $(env | grep -v "_LOCAL_IMAGE=" | grep "_IMAGE=" | grep -o "^[^=]*") ; do
-    IMAGE=${!IMAGE_VAR}
-    #shellcheck disable=SC2086
-    IMAGE_NAME="${IMAGE##*/}"
-    LOCAL_IMAGE="${REGISTRY}/localimages/$IMAGE_NAME"
-    if [[ $IMAGE == *"$CONTAINER_REGISTRY"* ]]; then
-      #shellcheck disable=SC2116
-      IMAGE="$(echo "${IMAGE/$CONTAINER_REGISTRY/$REPO_IMAGE_PREFIX}")"
-      echo "Image $IMAGE will be replaced with local image $LOCAL_IMAGE in image kustomization"
-    fi
-    sed -i -E "s $IMAGE$ $LOCAL_IMAGE g" "$FILE_PATH"
-  done
-}
-
-#
-# Create the BMO deployment (used for v1a4 only)
+# Create the BMO deployment (not used for CAPM3 v1a4 since BMO is bundeled there)
 #
 function launch_baremetal_operator() {
   pushd "${BMOPATH}"
 
   # Deploy BMO using deploy.sh script
 
   # Update container images to use local ones
-  cp "${BMOPATH}/config/manager/manager.yaml" "${BMOPATH}/config/manager/manager.yaml.orig"
```

Comment:
why are we  removing this and lines similar to these ? The idea was to leave the repos intact aftern the image manipulations we are doing.

Reply:
Basically I think it is unnecessary and makes it harder to understand what is happening. I added in the description this:

Backup and restore of some files before and after changes has
been removed. The reason for this is twofold:

1. All changed files are anyway restored every time the
    03_launch_mgmt_cluster.sh script is run, since it does a fresh clone
    every time, so manually doing these backups and restorations is not
    necessary.
2. Restoring the files directly when they have been used hides the true
    state for the user and makes it harder to understand what is happening.

If we still want to restore the repo I would much prefer to do it with `git reset --hard` at the end to avoid any mistakes.

Reply:
Well, if you want to know what image is the current deployment running, you should see the deployment in cluster, not the original files. Secondly, this change is not related to the fix you are doing, so I would rather keep it as it is for now and see if your changes are working properly or not first. We can take this part later if necessary IMO.

Reply:
Done